src/test/ui/did_you_mean/issue-46836-identifier-not-instead-of-negation.stderr

```diff
@@ -28,7 +28,10 @@ error: expected `{`, found `;`
 LL |     if not  // lack of braces is [sic]
    |     -- this `if` statement has a condition, but no block
 LL |         println!("Then when?");
-   |                               ^ expected `{`
+   |                               ^
+   |                               |
+   |                               expected `{`
+   |                               help: try placing this code inside a block: `{ ; }`
```

Comment:
This is a bit unfortunate.

Reply:
It is unfortunate, I'm not sure how to get around that (or if it's worth it to)

Reply:
Maybe only run the lint when there are no parse errors?

Reply:
I think the previous error message about using `!` for logical negation makes it clear what's wrong here, so this slightly odd suggestion isn't a problem.